lib/cfn-nag/cfn_nag_executor.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'trollop'
+require 'cfn-nag/cli_options'
+require 'cfn-nag/cfn_nag_config'
+
+class CfnNagExecutor
+  def initialize(require_input_path: false)
+    @require_input_path = require_input_path
+    @profile_definition = nil
+    @blacklist_definition = nil
+    @parameter_values_string = nil
+  end
+
+  def scan(aggregate_output: true,
```

Comment:
the require_input_path and aggregate_output are giving me a little trouble.  i know what they're there for.... but i'm wondering if you turn the other scan args into factory methods.... er do you need both? or could you have them both be on the scan() instead of split between the constructor and scan()?  there are basically two outcomes - single or aggregate, and only one of them requires the input path?